freeipa/src/main/java/com/sequenceiq/freeipa/service/SecurityConfigService.java

```diff
@@ -43,24 +57,41 @@ public SecurityConfig findOneByStack(Stack stack) {
                 if (!saltSecurityConfig.getSaltBootSignPrivateKey().equals(saltSecurityConfig.getSaltBootSignPrivateKeyVault())) {
                     saltSecurityConfig.setSaltBootSignPrivateKeyVault(saltSecurityConfig.getSaltBootSignPrivateKey());
                 }
-                if (!saltSecurityConfig.getSaltSignPrivateKey().equals(saltSecurityConfig.getSaltPasswordVault())) {
+                if (!saltSecurityConfig.getSaltPassword().equals(saltSecurityConfig.getSaltPasswordVault())) {
                     saltSecurityConfig.setSaltPasswordVault(saltSecurityConfig.getSaltPassword());
                 }
                 if (!saltSecurityConfig.getSaltSignPrivateKey().equals(saltSecurityConfig.getSaltSignPrivateKeyVault())) {
                     saltSecurityConfig.setSaltSignPrivateKeyVault(saltSecurityConfig.getSaltSignPrivateKey());
                 }
-                saltSecurityConfig = disabledSaltSecurityConfigRepository.save(saltSecurityConfig);
+                saltSecurityConfig = saltSecurityConfigRepository.save(saltSecurityConfig);
                 securityConfig.setSaltSecurityConfig(saltSecurityConfig);
             }
         }
         return securityConfig;
     }
 
+    public void createIfDoesntExists(Long stackId) throws TransactionExecutionException {
+        Stack stack = stackService.getStackById(stackId);
+        if (stack.getSecurityConfig() == null) {
+            LOGGER.debug("Create SecurityConfig for stack {}", stack.getResourceCrn());
+            SecurityConfig securityConfig = measure(() -> tlsSecurityService.generateSecurityKeys(stack.getAccountId()), LOGGER,
+                    "Generating security keys took {} ms for {}", stack.getName());
+            transactionService.required(() -> {
+                SecurityConfig savedSecurityConfig = securityConfigRepository.save(securityConfig);
+                stack.setSecurityConfig(savedSecurityConfig);
+                stackService.save(stack);
+            });
+        } else {
+            LOGGER.debug("SecurityConfig for stack {} already exists", stack.getResourceCrn());
+        }
+    }
+
     public void changeSaltPassword(Stack stack, String password) {
         SaltSecurityConfig saltSecurityConfig = Optional.ofNullable(stack.getSecurityConfig())
                 .map(SecurityConfig::getSaltSecurityConfig)
                 .orElseThrow(() -> new IllegalStateException("Stack " + stack.getResourceCrn() + " does not yet have a salt security config"));
         saltSecurityConfig.setSaltPassword(password);
-        disabledSaltSecurityConfigRepository.save(saltSecurityConfig);
+        saltSecurityConfig.setSaltPasswordVault(password);
```

Comment:
is it a fix for a current bug?

Reply:
I would say, yes, this should have been added in the first place, but slipped through review